noir-projects/aztec-nr/aztec/src/utils/collapse_array.nr

```diff
@@ -18,7 +16,7 @@ pub fn collapse_array<T, let N: u32>(input: [Option<T>; N]) -> BoundedVec<T, N>
     collapsed
 }
 
-fn verify_collapse_hints<T, let N: u32>(
+pub(crate) fn verify_collapse_hints<T, let N: u32>(
```

Comment:
How did you realize we don't need this outside of the crate?

Reply:
Oh, where it was called it said it was not visible because it's private. I thought about making it public, but given that it was private I at least made it pub(crate). But maybe it should be public?

Reply:
Oh, also, we recently implemented a "unused function" warnings. Maybe it's that?

Reply:
This is only public because it's used in a test. But now I'm thinking that the tests should be a submodule of this file so that we don't need to export this at all. Maybe this is a bad refactor.